aws/kinesis/core/shard_map.cc

```diff
@@ -174,24 +192,38 @@ void ShardMap::update_fail(const std::string &code, const std::string &msg) {
   backoff_ = std::min(backoff_ * 3 / 2, max_backoff_);
 }
 
-
 void ShardMap::clear_all_stored_shards() {
   end_hash_key_to_shard_id_.clear();
-  open_shard_ids_.clear();
-}
-
-void ShardMap::store_open_shard(const uint64_t shard_id, const uint128_t end_hash_key) {
-  end_hash_key_to_shard_id_.push_back(
-      std::make_pair(end_hash_key, shard_id));
-  open_shard_ids_.push_back(shard_id);
+  open_shards_.clear();
 }
 
 void ShardMap::sort_all_open_shards() {
   std::sort(end_hash_key_to_shard_id_.begin(),
```

Comment:
can we make it deterministic, so that every sort shards with the same hasEnd will be sorted in the same way?

https://en.cppreference.com/w/cpp/algorithm/sort
_The order of equal elements is not guaranteed to be preserved._

Reply:
The order of the element might not be important in this use case. std:stable_sort should be able to achieve deterministic order but it uses more memory and likely have higher time complexity deepening on the memory availability. It might not be a real concern for us but I think we can stick with this for now as this is what KPL has been using.